python/mxnet/module/base_module.py

```diff
@@ -511,9 +514,12 @@ def fit(self, train_data, eval_data=None, eval_metric='acc',
                 self.forward_backward(data_batch)
                 self.update()
                 try:
-                    # pre fetch next batch
-                    next_data_batch = next(data_iter)
-                    self.prepare(next_data_batch, sparse_row_id_fn=sparse_row_id_fn)
+                    if epoch_size and (nbatch == epoch_size - 1):
```

Comment:
Please correct me if I'm wrong, but this would be an infinite loop if the epoch_size was negative or 0, right? To be resilient against mis-usage (maybe coming from automated scripts), please assert ```epoch_size``` is positive and add a test accordingly

Reply:
No, this is an argument that allows early stopping of loop in a way because different machines can have different number of samples. It can't run into an infinite loop.

Reply:
This is also not really an easily testable scenario because the issue only comes up in distributed settings. The core logic just uses the argument provided. Rest of the changes in the PR is just for an example. I can add an assert sure

Reply:
I see, thanks for elaborating :)